services/navigation/builtin/builtin.go

```diff
@@ -220,90 +218,52 @@ func (svc *builtIn) SetMode(ctx context.Context, mode navigation.Mode, extra map
 	return nil
 }
 
-func (svc *builtIn) computeCurrentBearing(ctx context.Context, path []*geo.Point) (float64, error) {
-	props, err := svc.movementSensor.Properties(ctx, nil)
-	if err != nil {
-		return 0, err
-	}
-	if props.CompassHeadingSupported {
-		return svc.movementSensor.CompassHeading(ctx, nil)
-	}
-	pathLen := len(path)
-	return fixAngle(path[pathLen-2].BearingTo(path[pathLen-1])), nil
-}
-
 func (svc *builtIn) startWaypoint(extra map[string]interface{}) error {
 	svc.activeBackgroundWorkers.Add(1)
 	utils.PanicCapturingGo(func() {
 		defer svc.activeBackgroundWorkers.Done()
 
-		path := []*geo.Point{}
-		for {
-			if !utils.SelectContextOrWait(svc.cancelCtx, 500*time.Millisecond) {
-				return
-			}
-			currentLoc, _, err := svc.movementSensor.Position(svc.cancelCtx, extra)
+		currentLoc, err := svc.Location(svc.cancelCtx, extra)
+		if err != nil {
+			svc.logger.Errorw("failed to get gps location", "error", err)
+		}
+
+		navOnce := func(ctx context.Context) error {
+			bearingToGoal, _, err := svc.waypointDirectionAndDistanceToGo(ctx, currentLoc)
```

Comment:
Pretty sure this function can get removed too in favor of just using the `geo.BearingTo` function